cli/compose/loader/full-struct_test.go

```diff
@@ -18,6 +20,8 @@ func fullExampleConfig(workingDir, homeDir string) *types.Config {
 }
 
 func services(workingDir, homeDir string) []types.ServiceConfig {
+	configsPath, _ := crosspath.ParsePathWithDefaults(homeDir + "/configs")
```

Comment:
Hmm, that is an interesting feature when you want explicit control of the path elements yes. Does not apply to this specific case though (and would require some unit-testing to validate that everything works well in combination with the rest of the features).

Reply:
Actually, by doing the concatenation before trying the path categorization heuristics, you give more context to the heuristics to make the good decision: for example, by default, `\$foo` would get categorized as a `windows absolute from current drive` because of the leading `\`. However, `~/data/\$foo` gives more context, and the heuristic might detect correctly an home rooted linux path (not tried it, not 100% sure though)